pymc_marketing/mmm/multidimensional.py

```diff
@@ -1426,6 +1433,158 @@ def sample_posterior_predictive(
+        dist : pm.Distribution, optional
+            The distribution to use for the likelihood, by default pm.Gamma
+        name : str, optional
+            The name of the likelihood of the lift test contribution(s),
+            by default "lift_measurements". Name change required if calling
+            this method multiple times.
+
+        Raises
+        ------
+        RuntimeError
+            If the model has not been built yet.
+        KeyError
+            If the 'channel' column or any of the model dimensions is not present
+            in df_lift_test.
+
+        Examples
+        --------
+        Build the model first then add lift test measurements.
+
+        .. code-block:: python
+
+            import pandas as pd
+            import numpy as np
+
+            from pymc_marketing.mmm import (
+                GeometricAdstock,
+                LogisticSaturation
+            )
+
+            from pymc_marketing.mmm.multidimensional import MMM
+
+            model = MMM(
+                date_column="date",
+                channel_columns=["x1", "x2"],
+                target_column = "target",
+                adstock=GeometricAdstock(l_max=8),
+                saturation=LogisticSaturation(),
+                yearly_seasonality=2,
+                dims=("geo",),
+            )
+
+            X = pd.DataFrame({
+                "date": np.tile(pd.date_range(start="2025-01-01", end="2025-05-01", freq="W"), 2),
+                "x1": np.random.rand(34),
+                "x2": np.random.rand(34),
+                "target": np.random.rand(34),
+                "geo": 17 * ["FIN"] + 17 * ["SWE"]
+            })
+            y = X["target"]
+
+            model.build_model(X.drop(columns=["target"]), y)
+
+            df_lift_test = pd.DataFrame({
+                "channel": ["x1", "x1"],
+                "geo": ["FIN", "SWE"],
+                "x": [1, 1],
+                "delta_x": [0.1, 0.2],
+                "delta_y": [0.1, 0.1],
+                "sigma": [0.1, 0.1],
+            })
+
+            model.add_lift_test_measurements(df_lift_test)
+
+        """
+        if not hasattr(self, "model"):
+            raise RuntimeError(
+                "The model has not been built yet. Please, build the model first."
+            )
+
+        if "channel" not in df_lift_test.columns:
+            raise KeyError(
+                "The 'channel' column is required to map the lift measurements to the model."
+            )
+
+        for dim in self.dims:
+            if dim not in df_lift_test.columns:
+                raise KeyError(
+                    f"The {dim} column is required to map the lift measurements to the model."
+                )
+
+        # Define a target transformer from the target scaler.
+        def target_transform(target):
+            return target / self.scalers._target.to_numpy()
+
+        # Define a channel transformer from the channel scaler.
+        def channel_transform(channels):
```

Comment:
Based on the latest comment, can't you just wrap these in xarray, do the operation, then transform back to numpy? We will know all of the dimensions of these variables from the class and can use them in this local function

However, we can change the functions in lift_tests to be better structured if needed

Reply:
nice. Thanks